### DISCUSSION TITLE
Gatsby Deployment Issue

### DISCUSSION THREAD

**Discussion by bipin5** [gatsbyjs/gatsby#37657] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

I have a javascript file hubspot.js on src/api folder of my gatsby project which should become functions automatically with paths based on their file name (according to [Gatsby Documentation](https://www.gatsbyjs.com/docs/reference/gatsby-project-structure/)).

This API is called when submitting one form which should create a contact in hubspot. When I run the project in development, it successfully returns the response and contact is created. But, while using "serve public" to test in production, it returns:

POST http://localhost:3000/api/hubspot 404 (Not Found) and also in deploying using nginx configuration, it returns 405 error.

I also have another api function called status which is like this and is a GET api.

"export default function handler(req, res) { res.status(200).json({ status: 'ok' }); }"

Even this function is not called and returns not found error.

I don't have any backend as gatsby can handle that part by itself, as described here.

"Functions let you build dynamic applications without running servers. Submit forms, authenticate users, securely connect to external services, build GraphQL/REST APIs, and more. [Gatsby Function](https://www.gatsbyjs.com/docs/reference/functions/)
"

So, even in production, this must be handled and the api should be called when the form submits or when I make a GET request to /api/status. But, all I am getting is 404 not found and 405 error.

### Reproduction Link

https://github.com/bipin5/gatsby-build

### Steps to Reproduce

1. Create a Gatsby Project
2. Create a function in src/api folder as hello.js:
export default function handler(req, res) { 
  res.status(200).json({ status: 'ok' }); 
}
4. run gatsby build to build the project and "serve public" to run it as production.
5. It will then run on localhost:3000 (in my case). When you make a GET request to localhost:3000/api/hello, it will return a 404 Not found error
...

### Expected Result

I expected the output to be:
{
  status: 'ok'
}

### Actual Result

POST http://localhost:3000/api/hubspotContact 404 (Not Found)

### Environment

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1135G7 @ 2.40GHz
  Binaries:
    Node: 19.3.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.2.0 - C:\Program Files\nodejs\npm.CMD
  Languages:
    Python: 3.11.1
    gatsby-plugin-sitemap: ^6.6.0 => 6.6.0
    gatsby-remark-autolink-headers: ^6.6.0 => 6.6.0
    gatsby-remark-code-buttons: ^2.0.4 => 2.0.7
    gatsby-remark-images: ^7.6.0 => 7.6.0
    gatsby-source-filesystem: ^5.6.0 => 5.6.0
    gatsby-transformer-remark: ^6.6.0 => 6.6.0
    gatsby-transformer-sharp: ^5.6.0 => 5.6.0
    gatsby-transformer-yaml: ^5.6.0 => 5.6.0
```

### Config Flags

 flags: {
    DEV_SSR: false,
    FAST_DEV: true,
    DETECT_NODE_MUTATIONS: true,
    PRESERVE_FILE_DOWNLOAD_CACHE: true,
    PARALLEL_SOURCING: false,
    DETECT_NODE_MUTATIONS: false,
    PARTIAL_HYDRATION: false,
  },

**Comment by LekoArts** [comment]
Hi!

Your backend / your nginx server must handle these serverless functions, it's not something Gatsby (the framework) will handle for you. Currently the docs also don't state that. You need to handle those routes yourself.

Our setup in `gatsby serve` looks like this: https://github.com/gatsbyjs/gatsby/blob/bc6c3efb139b09576e12ce4b4cde8e61b148f86c/packages/gatsby/src/commands/serve.ts#L159-L183

**Reply by bipin5** [reply]
@LekoArts Can you help me on how I can configure my nginx server to handle the serverless functions? I have not been able to find a right way to do that. Any help would be appreciated. Thank you.
(I don't have a backend currently.)
